CHANGELOG.md

```diff
@@ -1,4 +1,20 @@
 # Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
+
+## [Version 0.4.0 (Public Preview)](https://github.com/awslabs/aws-mysql-jdbc/releases/tag/0.4.0) - 2021-12-14
+
+### Added
+* Connection plugin support through Connection Plugin Manager. More details can be found [here](https://github.com/awslabs/aws-mysql-jdbc#connection-plugin-manager).
+* [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring) for more sophisticated connection failure detection.
+* Tutorial on writing custom connection plugin can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).
+
+### Improvements
```

Comment:
Lets make sure our changelog actually conforms to this format. https://keepachangelog.com/en/1.0.0/
While we are at it lets fix the previous ones too.